### PR TITLE
fix errors reported by ruff

### PR DESCRIPTION
**Pull request by karmacoma-eth**

### CODE REVIEW

File: src/halmos/calldata.py [hunk 1]
```diff
@@ -189,7 +190,9 @@ def encode_tuple(self, items: List[EncodingResult]) -> EncodingResult:
         # See https://docs.soliditylang.org/en/latest/abi-spec.html#formal-specification-of-the-encoding
 
         # compute total head size
-        head_size = lambda x: x.size if x.static else 32
+        def head_size(x):
+            return x.size if x.static else 32
+
```
Comment (daejunpark) [inline src/halmos/calldata.py:195]: any pointer to the rationale of this?

File: src/halmos/cheatcodes.py [hunk 2]
```diff
@@ -643,9 +681,10 @@ def handle(sevm, ex, arg: ByteVec, stack, step_id) -> ByteVec | None:
 
         elif funsig == hevm_cheat_code.label_sig:
             addr = extract_bytes(arg, 4, 32)
-            label = extract_string_argument(arg, 1)
 
             # TODO: no-op for now
+            # label = extract_string_argument(arg, 1)
+
```
Comment (daejunpark) [inline src/halmos/cheatcodes.py:687]: does ruff do the deadcode elimination as well? that'd be quite amazing but also sound scary.

File: pyproject.toml [hunk 3]
```diff
@@ -41,3 +41,16 @@ target-version = ["py311", "py312"]
...
+    "E",  # pycodestyle
+    "F",  # pyflakes
+    "UP", # pyupgrade
+    "B",  # flake8-bugbear
+    "SIM",# flake8-simplify
+    "I",  # isort
+]
+ignore = [
+    "E501",  # line too long
+]
```
Comment (daejunpark) [inline pyproject.toml:56]: how to run this? no need to set up pre-commit or ci?

File: src/halmos/__main__.py [hunk 4]
```diff
@@ -11,33 +12,89 @@
...
+    ModelRef,
+    ZeroExt,
+    is_app,
+    is_bv,
+    sat,
+    set_option,
+    unknown,
+    unsat,
+)
+
```
Comment (daejunpark) [inline src/halmos/__main__.py:39]: nice!

**Review by daejunpark — APPROVED** [review]

File: pyproject.toml (hunk 3 above)
Comment (karmacoma-eth) [reply pyproject.toml:56]: currently running it manually with `ruff check src/ --fix`, we can look into adding to ci/pre-commit

File: src/halmos/cheatcodes.py (hunk 2 above)
Comment (karmacoma-eth) [reply src/halmos/cheatcodes.py:687]: it did point out that we defined `label` but never used it -- it doesn't automatically eliminate it

File: src/halmos/__main__.py [hunk 5]
```diff
@@ -1259,21 +1316,18 @@ def is_model_valid(model: AnyModel) -> bool:
...
                     return False
         return True
 
     # z3 model object
     else:
-        for decl in model:
-            if str(decl).startswith("f_evm_"):
-                return False
-        return True
+        return all(not str(decl).startswith("f_evm_") for decl in model)
```
[the remark this answers is not in the document]
Comment (karmacoma-eth) [reply src/halmos/__main__.py:1327]: `all()` does short-circuit, and the fix was suggested by ruff (but not automatically applied)

File: src/halmos/__main__.py [hunk 6]
```diff
@@ -320,8 +377,9 @@ def render_trace(context: CallContext, file=sys.stdout) -> None:
 
                 DeployAddressMapper().add_deployed_contract(target, contract_name)
                 addr_str = contract_name
-        except:
-            pass
+        except Exception:
+            # TODO: print in debug mode
+            ...
```
[the remark this answers is not in the document]
Comment (karmacoma-eth) [reply src/halmos/__main__.py:382]: it's basically the same, the convention I've seen is that `pass` is used to mark that you're intentionally doing nothing, and `...` is used to indicate that it's a placeholder for something to be added later

**Pull request merged**